### PR TITLE
Deeplink users to membership opt in flow

### PR DESCRIPTION
**Pull request by matheusmendes-cb**
closes 
#838 
fixes 

## What changed? Why?

Created deeplink page for membership opt in flow, added callback logic to redirect to signup and back to flow after login and created success state component.

## How has it been tested?

- [x] Locally
- [ ] Vercel Preview Branch
- [ ] Unit test
- [ ] Functional test

### CODE REVIEW

File: src/app/[locale]/(homepageDialogDeeplink)/action/become-member/layout.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import { getAuthenticatedData } from '@/components/app/pageUserProfile/getAuthenticatedData'
+import { PageProps } from '@/types'
+import { setCallbackQueryString } from '@/utils/server/searchParams'
+import { USER_ACTION_DEEPLINK_MAP } from '@/utils/shared/urlsDeeplinkUserActions'
+
+import { SuccessState } from './sucessState'
+
+export const dynamic = 'force-dynamic'
+
+export default async function Layout({ params, children }: React.PropsWithChildren<PageProps>) {
```
Comment (travisbloom-cb) [inline src/app/[locale]/(homepageDialogDeeplink)/action/become-member/layout.tsx:14]: whats the advantage of decoupling layout from page here? Generally I've only used layout when it permeates across multiple pages.

File: src/app/[locale]/(homepageDialogDeeplink)/action/become-member/sucessState.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  return (
+    <div className={cn('w-full space-y-6 text-center', dialogContentPaddingStyles)}>
+      <div>
+        <div className="inline-block rounded-full bg-green-600 p-1 text-white">
+          <Check className="h-6 w-6" />
+        </div>
+      </div>
+      <PageTitle size="sm">Thanks for becoming a 501(c)4 member of Stand With Crypto.</PageTitle>
+      <PageSubTitle withoutBalancer>
+        You made history by joining the largest pro-crypto organization in the U.S. Don’t stop here
```
Comment (travisbloom-cb) [inline src/app/[locale]/(homepageDialogDeeplink)/action/become-member/sucessState.tsx:26]: 🙌

File: src/app/[locale]/(homepageDialogDeeplink)/action/become-member/layout.tsx (hunk 1 above)
Comment (matheusmendes-cb) [reply src/app/[locale]/(homepageDialogDeeplink)/action/become-member/layout.tsx:14]: Layout is a server component here, so I can make request on server side for the user data and than render just the page as a client side application. I do it to minimize the amount of JS we send to the client as much as possible.

**Review by travisbloom-cb — DISMISSED** [review]

File: src/app/[locale]/(homepageDialogDeeplink)/action/become-member/layout.tsx (hunk 1 above)
Comment (matheusmendes-cb) [reply src/app/[locale]/(homepageDialogDeeplink)/action/become-member/layout.tsx]: Agreed, makes sense. Just fixed it.

**Review by travisbloom-cb — APPROVED** [review]

**Pull request merged**